src/components/TodoApp.jsx

```diff
@@ -0,0 +1,156 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import { TodoList } from './TodoList';
+import { TodosFilter } from './TodosFilter';
+
+export const TodoApp = React.memo(() => {
+  const [input, setInput] = useState('');
```

Comment:
it's better to move low lever input logic to separate component and just pass 'createTodo' method there as prop. You can avoid redundant rerenders this way, cause this input will rerender all components tree every time user changes single letter

Reply:
not fixed